src/main/java/frc/robot/commands/deployer/RaiseDeployer.java

```diff
@@ -27,12 +30,20 @@ public void initialize() {
     @Override 
     public boolean isFinished() {
         //Check is timer has passed timeout point or if deployer has reached limit switch
-        return (timer.hasElapsed(MOTOR_RUN_TIME) || deployer.isDeployerFowardLimitSwitchClosed());
+        if (deployer.isDeployerFowardLimitSwitchClosed()) {
+            return true;
+        }
+        else if (timer.hasElapsed(MOTOR_RUN_TIME)) {
+            timeoutCounter.increaseTimeoutCount();
+            return true;
+        }
+        return false;
     }
 
     @Override
     public void end(boolean interrupted) {
         deployer.setDeployerMotorSpeed(0);
+        timeoutCounter.increaseTimeoutCount();
```

Comment:
You should not also increase the timeoutCounter in end() when you already increased it in isFinished()

Reply:
I have made these changes.